lib/Serialization/Deserialization.cpp

```diff
@@ -3130,15 +3131,23 @@ class DeclDeserializer {
       attrs.setRawAttributeChain(DAttrs);
     }
 
-    auto overridden = MF.getDeclChecked(overriddenID);
-    if (!overridden) {
+    Expected<Decl *> overriddenOrError = MF.getDeclChecked(overriddenID);
+    Decl *overridden;
+    if (overriddenOrError) {
+      overridden = overriddenOrError.get();
+    } else {
+      if (overriddenAffectsABI || !ctx.LangOpts.EnableDeserializationRecovery) {
```

Comment:
Could we set `overridden = nullptr` if we're allowing compiler errors in this case (ie. `MF.allowCompilerErrors`)?

Reply:
Sure 👍